### PR TITLE
feature / stacked column charts

### PR DESCRIPTION
**Pull request by freeseus**
npm run tdd has a bunch of fails - but those are from master

If you switch to master and do the test, you'll see the same fails.

Should be safe to officially merge this with master.

### CODE REVIEW

File: src/scripts/util/data.model.js [hunk 1]
```diff
@@ -79,6 +79,39 @@ function valueDomain(model) {
...
+        }
+    }
+    vals.push(sum);
+});
+
+domain = d3.extent(vals);
+domain[0] > 0 ? domain[0] = 0 : 0; //if we do a chart where we don't want the start of the Y-axis to be 0, we'll want to be able to specify a lowest value other than 0
+}
+return domain;
+}
```
Comment (peter-mouland) [inline src/scripts/util/data.model.js]: When trying to understand this i noticed the hard-coded `date` value. should this be `model.x.series.key`?

I had a go at refactoring it to try and answer my next question around what the for loop with the conditional statement is trying to achieve.  Does the below make sense? i just pulled the loop+if nesting into its own function to help with self-documentation.

```javascript
function sumStackedValues(model){
    var extents = [];
    model.data.map(function (d, j) {
        var key, sum = 0;
        for (key in d.values[0]) {
            if (key !== 'date') {
                sum += d.values[0][key];
            }
        }
        extents.push(sum);
    });
    return extents;
}

function valueDomain(model){
    if(model.valueDomain){ return model.valueDomain; }

    var extents = (model.type === 'stacked') ? sumStackedValues(model) : setExtents(model);
    var domain = d3.extent(extents);
    if(!model.falseOrigin && domain[0] > 0){
        domain[0] = 0;
    }
    return domain;
}
```

**Comment by peter-mouland** [comment]
hey, none of the tests are failing on master, maybe your code isn't up-to-date?  [CircleCI history of master](https://circleci.com/gh/ft-interactive/o-charts/tree/master).

I think it is failing because a test example has been removed from the column demo page.  If you take a look at live (http://ft-interactive.github.io/o-charts/column-chart.html) you will spot one missing.  This can happen with complicated merge's.

Also I don't think `npm test` has been run on the latest code after the merge as it points out duplicate declarations within one of the updated functions.

**Comment by peter-mouland** [comment]
hey @freeseus i've gone through and made a few changes, I think it would be worth taking a look at the individual commits (7 of them) in the PR to see what i've done and if you agree with them or not.
@StuartFT do you want to take a look too?
